functionary/core/api/v1/views/task.py

```diff
@@ -67,8 +80,22 @@ def get_serializer_class(self):
         },
         parameters=HEADER_PARAMETERS,
     )
-    def create(self, request, *args, **kwargs):
-        request_serializer = self.get_serializer(data=request.data)
+    def create(self, request: Request, *args, **kwargs):
+        if not request.data.get("parameters"):
+            request.data["parameters"] = {}
+        else:
+            request.data["parameters"] = json.loads(request.data["parameters"])
+
+        # Wrap items in list to avoid dictionary changed size error
+        for param_name, _ in list(request.FILES.items()):
+            _handle_file_parameters(request, param_name)
+
+        request.data["parameters"] = json.dumps(request.data["parameters"])
```

Comment:
This further confuses me because you `json.loads` this data a few lines earlier, only to `json.dumps` it again here...

Reply:
The `json.loads` was needed to turn the json string into a dict so I could manipulate the `parameters` field. Then the `parameters` field is supposed to be in a valid json string, so I `json.dumps` it afterwards